library/std/src/thread/mod.rs

```diff
@@ -717,6 +717,13 @@ where
 
 /// Gets a handle to the thread that invokes it.
 ///
+/// # Panics
+///
+/// Panics if called beyond the end of `main`, when the Rust standard library's
```

Comment:
Why not just remove the guarantee?
```suggestion
/// This may panic if called beyond the end of `main`, if the Rust standard library's
```